Convert scripts to WP-CLI commands

rmccue opened the pull request:
Fixes #23.

Most of the commands I've imported as essentially the same as previous, so that you can change from `php whatever-script.php` to `wp glotpress whatever-script`. The various translation set-related scripts (export, import, and recheck-warnings) are now subcommands of `wp glotpress translation-set` however.

Eventual plan should be to replace these legacy scripts completely, but I think they really need a proper WP-CLI-oriented rewrite using subcommands properly.

rmccue committed:
Rewrite basic scripts to WP-CLI commands

rmccue committed:
Convert translation set commands

rmccue committed:
Throw a deprecation notice when subclassing

bradt commented:
@rmccue Took a quick look and the approach looks good. Needs a merge before I can test though.

rmccue commented:
Closing, it needs a weird merge, so will open a new (fresh) PR.

rmccue commented:
Oh, I am incredibly dumb.

rmccue committed:
Merge branch 'wordpress-plugin-dbi' into wp-cli-scripts

Conflicts:
	scripts/branch-project.php
	scripts/export.php
	scripts/import-originals.php
	scripts/import.php
	scripts/recheck-warnings.php

bradt commented:
I will test next time I work on GP. If someone beats me to it, all the better.

ocean90 commented on gp-includes/cli/remove-multiple-currents.php:
```diff
@@ -11,7 +9,12 @@ function run() {
-					echo "Duplicate with original_id#$prev_original_id. Translation#$translation->id\n";
+					WP_CLI::line( sprintf(
+						/* translators: 1: original ID, 2: new ID */
+						__( "Duplicate with original_id#%s. Translation#%s\n", 'glotpress' )
```
Missing comma

ocean90 commented on gp-includes/cli/remove-multiple-currents.php:
```diff
@@ -1,9 +1,7 @@
+	public function __invoke() {
 		$sets = GP::$translation_set->all();
 		foreach( $sets as $set ) {
 			echo "Processing set#$set->id...\n";
```
Should use `WP_CLI::line()`

ocean90 commented on gp-includes/cli/translation-set.php:
```diff
@@ -0,0 +1,145 @@
+		$po = new PO();
+		$po->import_from_file( $args[2] );
+		$added = $translation_set->import( $po );
+		printf( _n( "%s translation were added\n", "%s translations were added\n", $added, 'glotpress' ), $added );
```
Should use `WP_CLI::line()`

ocean90 commented on gp-includes/cli/wporg2slug.php:
```diff
@@ -0,0 +1,32 @@
+			WP_CLI::error( sprintf( __( 'No slug match for %s.', 'glotpress' ), $wporg_slug ) );
+		}
+
+		echo $slug;
```
Should use `WP_CLI::line()`

ocean90 committed:
Merge branch 'wordpress-plugin-dbi' into wp-cli-scripts

ocean90 committed:
Use `WP_CLI::line()` for output messages.

ocean90 committed:
CLI: Use `GP_PATH . GP_INC` instead of `dirname( __FILE__ )`.

ocean90 committed:
CLI: Enhance output of some commands.

Also fix a fatal error in the `add-admin` command, `GP_User::by_login()` doesn't exist anymore.

ocean90 committed:
CLI: Add missing sprintf() call in import-originals
